testing/test_all_components.py

```diff
@@ -0,0 +1,112 @@
+"""
+This module contains code for auto-discovering Widget classes, instantiating
+them in a programatically defined view class as a new type, which is then
+instantiated and then testing behavior of the widget objects that appear
+to get instantiated upon access.
+"""
+
+from widgetastic.widget import View, Widget
+import widgetastic_patternfly as wp
+import inspect
+import pytest
+import re
+
+
+collected_widgets = {
+    name: t for name, t in inspect.getmembers(wp)
+    if (inspect.isclass(t) and issubclass(t, Widget)
+        and not issubclass(t, wp.ParametrizedView))}
+""" Subclasses of Widget from the widgetastic_patternfly module (including
+it's own imports), that are gonna be tested. Not including the
+ParametrizedView. """
+
+
+DUMMY_NAME = "name_of_the_dummy"
+DUMMY_ID = "id_of_the_dummy"
+DUMMY_LOCATOR = "/dummy"
+DUMMY_TREE_ID = "SOME_DUMMY_TREE_ID"
+
+
+def camel_to_snake(s):
```

Comment:
https://github.com/RedHatQE/widgetastic.core/blob/master/src/widgetastic/utils.py#L493

Consider using this `wt.core` util instead, if it will fit the use case here.

Reply:
The camel_to_snake dealt well with the spaced strings:
```
In [6]: re.sub(r'(?<!^)(?=[A-Z])', '_', "CamelToSSSNake").lower()                             
Out[6]: 'camel_to_s_s_s_nake'
```

But this test is not about attribute names, so I changed this as you suggested.